src/resolvers/Promotion.js

```diff
@@ -0,0 +1,267 @@
+  let payload = discountAmtConverter(promotion._doc);
+  payload.image = imageSizeConverter(promotion._doc.image);
+  promotion.promotionService.map((promotionService) => {
+    estimatedServiceTimeConverter(promotionService._doc);
+  });
+  payload.promotionService = promotion.promotionService;
+  return payload;
+};
+
+const discountAmtConverter = (promotion) => {
+  promotion.discountAmt = promotion.discountAmt.toString();
+  return promotion;
+};
+
+const imageSizeConverter = (imageStorage) => {
+  try {
+    imageStorage.imageSize = imageStorage.imageSize.toString();
+  } catch (error) {
+    console.error(error);
+  }
+  return imageStorage;
+};
+
+const estimatedServiceTimeConverter = (PromotionService) => {
+  try {
+    PromotionService.estimatedServiceTime = PromotionService.estimatedServiceTime.toString();
+  } catch (error) {
+    console.error(error);
+  }
+  return PromotionService;
+};
+
+//TODO: #60 @LUXIANZE Add auth check before appropriate query and mutations
+const PromotionResolver = {
+  Query: {
+    promotions: async (root, { filter }, context, info) => {
+      /**
+       * - only allow logged in user
+       * - check filter validity
+       * - retrive based on filter
+       */
+      if (filter === null) {
+        throw new UserInputError("Invalid filter");
+      }
+
+      /**
+       * check filter validity
+       */
+      let parsedFilter;
+      try {
+        parsedFilter = JSON.parse(filter);
+      } catch (error) {
+        throw new ApolloError(error, 500);
+      }
+
+      /**
+       * retrieve based on filter
+       */
+      let promotions = await Promotion.find(parsedFilter);
+      promotions.map((promotion) => {
+        promotionConverter(promotion);
+      });
+
+      return promotions;
+    },
+    promotion: async (root, { id }, context, info) => {
+      /**
+       * - only allow logged in user
+       * - check id validity
+       * - retrive based on id
+       */
+
+      /**
+       * check id validity
+       */
+      if (id === null) {
+        throw new UserInputError("No ID provided");
+      }
+
+      /**
+       * validate id format
+       */
+      if (!mongoose.Types.ObjectId.isValid(id))
+        throw new UserInputError(`${id} is having an invalid format`);
+
+      /**
+       * retrive based on id
+       */
+      let promotion = await Promotion.findById(id);
+      promotion = promotionConverter(promotion);
+      promotion.id = promotion._id;
+
+      return promotion;
+    },
+  },
+  Mutation: {
+    createPromotion: async (root, { promotionInput }, context, info) => {
+      /**
+       * - only allow logged in Brand Owner to create promotions
+       * - throw if id provided
+       * - check if the promotion name is used
+       * - check if the date range is valid
+       * - insert promotion
+       */
+
+      /**
+       * check user type is Brand owner
+       */
+      if (context.user.type !== USER_TYPE.BRANDOWNER) {
```

Comment:
only brand owner can create promotion?
manager of branch cannot?

Reply:
since promotion is brand scope, I only allow brand owner to create and delete. manager can modify only